packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -470,6 +473,57 @@ async def copy_objects_recursively(
             limit=_MAX_CONCURRENT_COPY,
         )
 
+    async def get_bytes_streamer_from_object(
+        self,
+        bucket_name: S3BucketName,
+        object_key: S3ObjectKey,
+        *,
+        chunk_size: int = DEFAULT_READ_CHUNK_SIZE,
+    ) -> BytesStreamer:
+        """stream read an object from S3 chunk by chunk"""
+
+        # NOTE `download_fileobj` cannot be used to implement this because
+        # it will buffer the entire file in memory instead of reading it
+        # chunk by chunk
```

Comment:
what about this?
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/customizations/s3.html#boto3.s3.transfer.TransferConfig
I see chunks configuration there. also I see multipart download and multi threading. Are you sure about this?

Reply:
Current implementation is still valid.
I tried really hard to use `download_fileobj`, but could not do so.
The `download_fileobj` writes to a `file object`. The options you are highlighting, help with tuning the chunks and parallelism of writing to the output. You cannot pause the process and resume it.

To do so, I had to mull the checks one at a time, like it's done here.

Reply:
mull?

Reply:
@sanderegg I'll rewrite the last part.

To achieve on demand file chunk download, the only way I found was to pull once chunk like I did here.